internal/sockets/dial.go

```diff
@@ -13,17 +13,17 @@ func Dial(rawAddr string) (int, error) {
 	opt := addr.Query()
 	noDelay := intopt(opt, "nodelay", 1)
 	if err = syscall.SetsockoptInt(fd, syscall.IPPROTO_TCP, syscall.TCP_NODELAY, noDelay); err != nil {
-		syscall.Close(fd)
+		Close(fd)
```

Comment:
Changed to use the `Close` function declared in the package which helped when tracking down where fd 0 was being closed.